### PR TITLE
Delete user messages

### PR DESCRIPTION
**Pull request by sjohnston00**

### CODE REVIEW

**Comment by sjohnston00** [comment]
Just fixed a tiny merge conflict with main from branch `fix-socket-connection`, make sure to pull changes from this branch

**Comment by sjohnston00** [comment]
> @sjohnston00 This implements the functionality however newly generated messages do not have an onclick function and I'm having a lot of trouble dealing with that.

### The problem
When the script loads, it then adds on click event listener to all the buttons, but then when a new button is added even though it has the same className it doesn't have the event listener.

### The solution
For now, I'm thinking that we add 2 functions:
1. For getting all the buttons and adding the event listeners.
2. For removing all the current event listeners on the message buttons (so as not to layer click functions on top of each other, kind of like a cleanup process before, applying the event listeners again).
Then when we get a new message, we run the cleanup function and then apply a whole new set of event listeners

**Comment by gumbogal** [comment]
That is pretty much the route I was going down though it was difficult to make any progress through the brain fog. I didn't intend to do the cleanup process because I thought it would _overwrite_ the initial onclick, but I could be wrong.

**Comment by sjohnston00** [comment]
> That is pretty much the route I was going down though it was difficult to make any progress through the brain fog. I didn't intend to do the cleanup process because I thought it would _overwrite_ the initial onclick, but I could be wrong.

Yeah I don’t know wether it overwrites it either, was just commenting because I don’t want the click event to have 5 function tied to it

File: src/content/scripts/tournamentMessaging.js [hunk 1]
```diff
@@ -4,20 +4,59 @@ const socket = io.connect(CONNECTION_STRING)
...
+        xmlns="http://www.w3.org/2000/svg"
+        width="28"
+        height="28"
+        fill="currentColor"
+        class="bi bi-x"
+        viewBox="0 0 16 16"
+      >
+        <path
+          d="M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z"
+        />
```
Comment (gumbogal) [inline src/content/scripts/tournamentMessaging.js:56]: What the fuck is going on?????

**Review by gumbogal — APPROVED** [review]

**Pull request merged**